src/Http/Request.php

```diff
@@ -213,4 +220,31 @@ public static function getBatchApi()
     {
         return static::OUTLOOK_ROOT_URL . static::OUTLOOK_VERSION . DIRECTORY_SEPARATOR . self::OUTLOOK_BATCH_ENDPOINT;
     }
+
+    public function setAccessToken(string $accessToken): void
+    {
+        $this->accessToken = $accessToken;
+    }
+
+    public function getNewHeader(string $url, array $params = []): array
+    {
+        $options = [
```

Comment:
I think you should be using the `refresh` method in Token class to get the new token entity instead of adding it here